mlflow/server/handlers.py

```diff
@@ -262,6 +262,43 @@ def wrapper(*args, **kwargs):
 ]
 
 
+def _disable_unless_serving_artifacts(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        from mlflow.server import SERVE_ARTIFACTS_ENV_VAR
+
+        if not os.environ.get(SERVE_ARTIFACTS_ENV_VAR):
+            return Response(
+                (
+                    "Endpoints for mlflow artifacts service are disabled. To enable them, "
+                    "run `mlflow server` with `--serve-artfiacts-opt`"
+                ),
```

Comment:
```suggestion
                (
                    "Endpoint disabled due to the mlflow server running without `--serve-artifacts`"
                    "To enable artifacts server functionality, run `mlflow server` with "
                    "`--serve-artifacts`"
                ),
```


Can we use a similar message as `_disable_mlflow_artifacts_only`?

Reply:
When we make an artifact request against mlflow server launched without `--serve-artifacts`, the error message looks like this:

```
raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 503 Server Error: SERVICE UNAVAILABLE for URL: 
http://localhost:5000/api/2.0/mlflow-artifacts/artifacts/experiments/0/15d993c8025f471a833dd97969920970/artifacts/a.txt
```

It doesn't explain why it's unavailable This is because we use `raise_for_status` to raise a request error:

https://github.com/mlflow/mlflow/blob/f6a6b20c1330a9c4808ad20ba88ec311a53fdc25/mlflow/store/artifact/http_artifact_repo.py#L28

We might need to use the approach in this StackOverflow answer: https://stackoverflow.com/a/52392219